cmd/dptp-pools-cm/main.go

```diff
@@ -95,28 +92,29 @@ func main() {
 	ctx := controllerruntime.SetupSignalHandler()
 	ctx, cancel := context.WithCancel(ctx)
 
-	kubeconfigChangedCallBack := func(e fsnotify.Event) {
-		logrus.WithField("event", e.String()).Info("Kubeconfig changed, exiting to get restarted by Kubelet and pick up the changes")
+	kubeconfigChangedCallBack := func() {
+		logrus.Info("Kubeconfig changed, exiting to get restarted by Kubelet and pick up the changes")
 		cancel()
 	}
 
-	kubeconfigs, err := util.LoadKubeConfigs(opts.kubeconfig, opts.kubeconfigDir, kubeconfigChangedCallBack)
+	kubeconfigs, err := opts.kubernetesOptions.LoadClusterConfigs(kubeconfigChangedCallBack)
 	if err != nil {
 		logrus.WithError(err).Fatal("failed to load kubeconfigs")
 	}
 	hiveConfig, hasHive := kubeconfigs[hiveContextName]
 	if !hasHive {
-		kubeconfigs[hiveContextName], err = rest.InClusterConfig()
+		inClusterConfig, err := rest.InClusterConfig()
+		if err != nil {
+			logrus.WithError(err).Fatal("failed to load in-cluster config")
+		}
+		kubeconfigs[hiveContextName] = *inClusterConfig
 		if err != nil {
 			logrus.WithError(err).Fatalf("--kubeconfig had no context for '%s' and loading InClusterConfig failed", hiveContextName)
 		}
 		logrus.Infof("Loaded %q context from in-cluster config", hiveContextName)
-		if err := util.WatchFiles([]string{"/var/run/secrets/kubernetes.io/serviceaccount/token"}, kubeconfigChangedCallBack); err != nil {
```

Comment:
This is the default path for SA's token.
If we still need this, we will fix it upstream.